src/PrestaShopBundle/Command/PrintCommandsAndQueriesForDocsCommand.php

```diff
@@ -0,0 +1,240 @@
+            )
+        ;
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     *
+     * @return int|null
+     */
+    public function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        $destinationDir = $this->getDestinationDir($input);
+
+        if (!$this->confirmExistingFileWillBeLost($destinationDir, $input, $output)) {
+            $output->writeln('<comment>Cancelled</comment>');
+
+            return null;
+        }
+
+        $this->filesystem->remove($destinationDir);
+
+        $definitions = $this->getCommandHandlerDefinitions();
+        ksort($definitions);
+
+        foreach ($definitions as $domain => $definitionsByType) {
+            $content = $this->twigEnv->render('src/PrestaShopBundle/Command/views/cqrs-commands-list.md.twig', [
+                'domain' => $domain,
+                'definitionsByType' => $definitionsByType,
+            ]);
+
+            $this->filesystem->dumpFile($this->getDestinationFilePath($destinationDir, $domain), $content);
+        }
+
+        $indexFileContent = $this->twigEnv->render('src/PrestaShopBundle/Command/views/cqrs-commands-index.md.twig');
+        $this->filesystem->dumpFile(sprintf('%s/_index.md', $destinationDir), $indexFileContent);
+        $output->writeln(sprintf('<info>dumped commands & queries to %s</info>', $destinationDir));
+
+        return 0;
+    }
+
+    /**
+     * @param string $targetDir
+     * @param string $domain
+     *
+     * @return string
+     */
+    private function getDestinationFilePath(string $targetDir, string $domain): string
+    {
+        return sprintf(
+            '%s/%s.md',
+            $targetDir,
+            $this->stringModifier->convertCamelCaseToKebabCase($domain)
+        );
+    }
+
+    /**
+     * @return array<string, array<string, array<int, CommandHandlerDefinition>>>
+     */
+    private function getCommandHandlerDefinitions(): array
+    {
+        $handlerDefinitions = $this->getContainer()->getParameter('prestashop.commands_and_queries');
+        /** @var CommandHandlerDefinitionParser $commandHandlerDefinitionParser */
+        $commandHandlerDefinitionParser = $this->getContainer()->get('prestashop.core.provider.command_handler_definition_parser');
+
+        $commandDefinitionsByDomain = [];
+        foreach ($handlerDefinitions as $handlerClass => $commandClass) {
+            $commandDefinition = $commandHandlerDefinitionParser->parseDefinition($handlerClass, $commandClass);
+            if ($commandDefinition->getType() === CommandHandlerDefinition::TYPE_QUERY) {
+                $commandDefinitionsByDomain[$commandDefinition->getDomain()][CommandHandlerDefinition::TYPE_QUERY][] = $commandDefinition;
+                continue;
+            }
+
+            $commandDefinitionsByDomain[$commandDefinition->getDomain()][CommandHandlerDefinition::TYPE_COMMAND][] = $commandDefinition;
```

Comment:
```suggestion
            $commandDefinitionsByDomain[$commandDefinition->getDomain()][$commandDefinition->getType()][] = $commandDefinition;
```
Simplification suggestion